server/app/uk/gov/ons/addressIndex/server/controllers/RandomController.scala

```diff
@@ -55,7 +55,7 @@ class RandomController @Inject()(val controllerComponents: ControllerComponents,
     val verb = verbose.flatMap(x => Try(x.toBoolean).toOption).getOrElse(false)
 
     val epochVal = epoch.getOrElse("")
-    val fromSourceVal = fromsource.getOrElse("all")
+    val fromSourceVal = {if (fromsource.getOrElse("all").isEmpty) "all" else fromsource.getOrElse("all")}
```

Comment:
This can be shortened using Scala to fromsource.filter(!_.isEmpty).getOrElse("all"). The braces are not needed. It is also a duplicate of code above.